snapcraft/storeapi/errors.py

```diff
@@ -105,6 +106,17 @@ def __init__(self, name, channel=None, arch=None, series=None):
         super().__init__(name=name, channel=channel, arch=arch, series=series)
 
 
+class NoSnapIdError(StoreError):
+
+    fmt = (
+        'Unable to get snap ID for snap {snap_name!r}: The store returned '
+        'data, but included no ID. This is an error in the store, please open '
+        'a new topic on the forum: {forum_url}')
```

Comment:
A better error message suggestion:

"Missing snap ID for {snap_name!r}. Please open a new topic in the store category on the forum: {forum_url}"

Can we link to `/c/store` directly?

Reply:
I reworked the error a little to link to the store category.